presidio-anonymizer/presidio_anonymizer/anonymizer_engine.py

```diff
@@ -43,9 +42,11 @@ def anonymize(self, engine_request: AnonymizerRequest) -> str:
         # get anonymizer type class for the analyzer result (replace, redact etc.)
         # trigger the anonymizer method on the section of the text
         # perform the anonymization
+
         # concat the anonymized string into the output string
         for analyzer_result in analyzer_results:
-            self.__validate_position_over_text(analyzer_result, text_len)
+            text_to_anonymize = text_manipulator.validate_and_get_text_in_position(
```

Comment:
Consider separating into two calls: validate and get text in position

Reply:
I moved the validation to another place which makes more sense and all validations are at.